milli/src/update/index_documents/transform.rs

```diff
@@ -147,7 +153,9 @@ impl Transform<'_, '_> {
 
             // We prepare the fields ids map with the documents keys.
             for (key, _value) in &document {
-                fields_ids_map.insert(&key).context("field id limit reached")?;
+                let field_id = fields_ids_map.insert(&key).context("field id limit reached")?;
+
+                *fields_ids_distribution.entry(field_id).or_default() += 1;
```

Comment:
```suggestion
                *fields_ids_distribution.entry(field_id).or_default() += 1u64;
```

Reply:
I'd rather suggest something like
```rust
let mut fields_ids_distribution: HashMap<_, u64> = HashMap::new();
```

Type inference still isn't possible in your case because of possible zero-sized `document` variable. I mean `for` loop may not happen as well as increment with `1u64` and then `V` type isn't known.

Reply:
No, it doesn't work like that, type resolution is done at compile-time, it doesn't depend on whether the documents `for` loop runs or not. But I am maybe misunderstanding something, here?

Reply:
Ok I think figured it out. It doesn't work that way
```rust
use std::collections::HashMap;

fn main() {
    let mut hashmap = HashMap::new();

    *hashmap.entry(1).or_default() += 1u64;
}
```

but does work that way
```rust
use std::collections::HashMap;

fn main() {
    let mut hashmap = HashMap::new();

    *hashmap.entry(1).or_insert(0) += 1;
}
```

I think that's the reason in the codebase were mentions of `or_insert(0)` construction and no `or_default()`.

Reply:
fixed

Reply:
Ho ok, yeah that's probably the reason why. Thank you for diving into this inference problem!